lib/get-distro.js

```diff
@@ -0,0 +1,65 @@
+// External Modules
+const fs = require('fs');
+const os = require('os');
+
+// Internal Modules
+const getOS = require('./get-os');
+const log = require('./utils').log;
+
+// Constants
+const DEFAULT_OS_PATH = '/etc/os-release';
+
+// Helper in Getting Dist from GetOS
+const getDist = (callback) => {
+    if (typeof callback !== 'function') {
+        throw new Error('Must provide a function callback');
+    }
+
+    let dist = {};
+
+    return getOS((error, result) => {
+        if (error) {
+            log(`Couldn't get OS information: ${error}`);
+        } else if (result) {
+            dist.os = result.os;
+            if (dist.os !== 'win32' && dist.os !== 'darwin') {
+                dist.name = `${result.dist} ${result.release}`;
+                dist.release = result.release;
+                dist.code = result.codename;
+            }
+        }
+
+        return callback(null, dist);
+    });
+};
+
+module.exports = (path, callback) => {
+    if (typeof path === 'function' && os.platform() !== 'win32') {
+        callback = path;
+        path = DEFAULT_OS_PATH;
+    }
```

Comment:
so here you are expecting 2 params, but you are trying to make `path` optional, however this is only in effect when `os.platform()` value is `win32` -- even conceptually this is strange

for node, because there is technically no function overloading, you are simply sniffing input params to support multiple function signatures and so you want to be very consistent in how you shift these arguments so you don't cause code breakage

if you break this down, you are actually changing how a function signature is interpreted based on the running context of the app -- which is really bad here

(I didn't expand the entirety of index.js so I could be mistaken but...) if we reference the caller to this from line:252 in index.js, this call is nested inside an if check for `os.platform() === 'win32'` yet here, we are only shifting the path when it isn't `win32`? Which means `path` at this point would hold the callback, and so it bypasses everything, tries to call `getDist()` with undefined, and will now just throw new Error -- but since this was requested by a (likely) async function, you are now hung because the callback function (which is actually in `path`) is never called

Reply:
I would remove `path` forever but this function is also used in testing but I did workaround - and thanks for showing the dumb/easy failure case! Loved it!